modules/uhttp/filters.go

```diff
@@ -48,76 +49,71 @@ func (f FilterFunc) Apply(ctx fx.Context, w http.ResponseWriter, r *http.Request
+	ext.HTTPUrl.Set(span, r.URL.String())
+	defer span.Finish()
+
+	ctx = ctx.(*fxcontext.Context).WithContextAwareLogger(span)
+	ctx = &fxcontext.Context{
+		Context: opentracing.ContextWithSpan(ctx, span),
+	}
+	r = r.WithContext(ctx)
+	next.ServeHTTP(ctx, w, r)
 }
 
 // authorizationFilter authorizes services based on configuration
-func authorizationFilter(host service.Host) FilterFunc {
-	return func(ctx context.Context, w http.ResponseWriter, r *http.Request, next Handler) {
-		fxctx := &fxcontext.Context{
-			Context: ctx,
-		}
+type authorizationFilter struct {
+	authClient auth.Client
+	scope      tally.Scope
+}
 
-		if err := host.AuthClient().Authorize(fxctx); err != nil {
-			host.Metrics().SubScope("http").SubScope("auth").Counter("fail").Inc(1)
-			fxctx.Logger().Error(auth.ErrAuthorization, "error", err)
-			w.WriteHeader(http.StatusUnauthorized)
-			fmt.Fprintf(w, "Unauthorized access: %+v", err)
-			return
-		}
-		next.ServeHTTP(fxctx, w, r)
+func (f authorizationFilter) Apply(ctx fx.Context, w http.ResponseWriter, r *http.Request, next Handler) {
+	if err := f.authClient.Authorize(ctx); err != nil {
+		f.scope.Counter("auth.fail").Inc(1)
+		ctx.Logger().Error(auth.ErrAuthorization, "error", err)
+		w.WriteHeader(http.StatusUnauthorized)
+		fmt.Fprintf(w, "Unauthorized access: %+v", err)
+		return
 	}
+	next.ServeHTTP(ctx, w, r)
 }
 
 // panicFilter handles any panics and return an error
 // panic filter should be added at the end of filter chain to catch panics
-func panicFilter(host service.Host) FilterFunc {
-	return func(ctx context.Context, w http.ResponseWriter, r *http.Request, next Handler) {
-		fxctx := &fxcontext.Context{
-			Context: ctx,
+type panicFilter struct {
+	scope tally.Scope
+}
+
+func (f panicFilter) Apply(ctx fx.Context, w http.ResponseWriter, r *http.Request, next Handler) {
+	defer func() {
+		if err := recover(); err != nil {
+			ctx.Logger().Error("Panic recovered serving request", "error", err, "url", r.URL)
+			f.scope.Counter("panic").Inc(1)
+			w.Header().Add(ContentType, ContentTypeText)
+			w.WriteHeader(http.StatusInternalServerError)
+			fmt.Fprintf(w, "Server error: %+v", err)
 		}
-		defer func() {
-			if err := recover(); err != nil {
-				fxctx.Logger().Error("Panic recovered serving request", "error", err, "url", r.URL)
-				host.Metrics().Counter("http.panic").Inc(1)
-				w.Header().Add(ContentType, ContentTypeText)
-				w.WriteHeader(http.StatusInternalServerError)
-				fmt.Fprintf(w, "Server error: %+v", err)
-			}
-		}()
-		next.ServeHTTP(fxctx, w, r)
-	}
+	}()
+	next.ServeHTTP(ctx, w, r)
+}
+
+// metricsFilter adds any default metrics related to HTTP
+type metricsFilter struct {
+	scope tally.Scope
+}
+
+func (f metricsFilter) Apply(ctx fx.Context, w http.ResponseWriter, r *http.Request, next Handler) {
+	defer f.scope.Tagged(map[string]string{modules.TagStatus: w.Header().Get("Status")}).Counter("total").Inc(1)
```

Comment:
Tally has its own benchmarks - https://github.com/uber-go/tally/. The only addition here is to extract header value from http, which is a map[string] get. I don't see any value adding beanchmark here, but we should benchmark overall modules performance.